### PR TITLE
feature/graceful-degradation-homepage

### PR DESCRIPTION
**Pull request by rav-pradhan**
### What

- Implemented graceful degradation states as per Sketch copies
- Created new partial for rendering when no main figures successfully loaded

### How to review

- Check that the changes make sense
- You can review in conjunction with the [frontend homepage controller PR](https://github.com/ONSdigital/dp-frontend-homepage-controller/pull/46) (follow graceful degradation steps below)

**Graceful degradation states for checking locally**
1. Comment out a main figure URI in `homepage.go` in dp-frontend-homepage-controller: that corresponding tile in the homepage should show a failed to load state
2. Comment out all main figure URIs in `homepage.go` in dp-frontend-homepage-controller: all the tiles are replaced by a single tile, aligned to the latest releases tile, that shares the same fail state message
3. Contrive a way to not have any latest releases in the last 7 days (not sure how best to do this locally). The second sentence in the latest releases tile should no longer be present (i.e., "we have published _n_ releases...")
4. Remove any featured content you have in your homepage `data.json` in you Zebedee content dir. The In Focus section should then not render at all

### Who can review

Anyone

### CODE REVIEW

File: assets/templates/homepage/main-figures.tmpl [hunk 1]
```diff
@@ -14,6 +14,7 @@
             <a href="https://www.ons.gov.uk/timeseriestool" class="tile__link">From our time series explorer</a>
         </span>
     </header>
+    {{ if .Data.HasMainFigures }}
```
Comment (dandwal) [inline assets/templates/homepage/main-figures.tmpl:17]: Noticed when I commented out multiple but not all URIs that there the tiles don't line up as they normally do (note the gap in the image). Again not introduced in this PR but whenever the last work on this was done it was possibly just missed. This is also not a mojor issue, low chance of people seeing this and next to no impact. So this is not blocking approval.

Screenshot 2020-10-21 at 14 01 31

File: assets/templates/homepage/main-figures-none.tmpl [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="col col--lg-29 col--md-29">
```
Comment (dandwal) [inline assets/templates/homepage/main-figures-none.tmpl:1]: Padding/Margin on Main figures in the medium viewport looks a little off
Screenshot 2020-10-21 at 14 06 10

File: assets/templates/homepage/main-figures-none.tmpl [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+    <article class="tile tile__content margin-top-lg--2 margin-top-md--2 margin-left-md--1 margin-left-lg--0 height-lg--31">
+        <div class="tile__text-description font-weight-700 margin-top--2 margin-bottom--2"> {{ localise "ExperiencingIssues" .Language 1}} </div>
+        <div class="tile__text-description"> {{ localise "RefreshOrVisitTimeseriesTool" .Language 1 | safeHTML }} </div>
```
Comment (dandwal) [inline assets/templates/homepage/main-figures-none.tmpl:4]: I think the link styling here doesn't match the rest of the page, no underline, underline appears on hover but should change colour, the focus is missing the yellow glow and thick underline. Focus state can be seen here
Screenshot 2020-10-21 at 14 19 49

Where it should look like:
Screenshot 2020-10-21 at 14 20 04

**Review by dandwal — COMMENTED** [review]
### Fully working working

- Release calendar
- Featured content

### Some issues around

- Main figures (see comments on PR for further details

### Notes

For future reference, you can test no releases by removing the contents of your .`../ZebedeeDirectory/master/releases`.

File: assets/templates/homepage/main-figures-none.tmpl (hunk 3 above)
Comment (rav-pradhan) [reply assets/templates/homepage/main-figures-none.tmpl:4]: Thanks for spotting this. I'd just copied over the pattern we used for the error message on individual tiles. The change I'll make for this should update it on those individual error messages too

File: assets/templates/homepage/main-figures.tmpl [hunk 4]
```diff
@@ -14,6 +14,7 @@
             <a href="https://www.ons.gov.uk/timeseriestool" class="tile__link">From our time series explorer</a>
```
[the remark this answers is not in the document]
Comment (rav-pradhan) [reply assets/templates/homepage/main-figures.tmpl:14]: It looks like small and medium viewports had the wrong value being passed to `DatePeriodFormat`. We have this for the desktop viewport on l26

`Aged 16 to 64 seasonally adjusted ({{ DatePeriodFormat $EmploymentRate.Date}})`

whereas it is this for lines 157 and 327 (sm/md):

`Aged 16+ seasonally adjusted ({{ DatePeriodFormat $EmploymentRate.Date }})`

This does mean the homepage can load again, although it appears there are some additional styling tweaks to be made. I will look into this as part of this work

File: assets/templates/homepage/main-figures.tmpl [hunk 5]
```diff
@@ -14,6 +14,7 @@
...
         </span>
```
[the remark this answers is not in the document]
Comment (rav-pradhan) [reply assets/templates/homepage/main-figures.tmpl:15]: I believe I found a fix for this - we have a utility class in Sixteens that aligns text to the top of its container. That appears to resolve this issue on lg viewports

File: assets/templates/homepage/main-figures.tmpl (hunk 1 above)
Comment (rav-pradhan) [reply assets/templates/homepage/main-figures.tmpl:17]: Looks like when we have the error message the height isn't growing. I suspect the employment/unemployment tile is even shorter when both fail to load. Probably fair to say it's not a major issue, but I will look into this a bit

File: assets/templates/homepage/main-figures.tmpl (hunk 5 above)
[the remark this answers is not in the document]
Comment (rav-pradhan) [reply assets/templates/homepage/main-figures.tmpl:15]: image

should be fixed in latest commits

File: assets/templates/homepage/main-figures.tmpl (hunk 4 above)
[the remark this answers is not in the document]
Comment (rav-pradhan) [reply assets/templates/homepage/main-figures.tmpl:14]: Should be fixed in latest commit

File: assets/templates/homepage/main-figures-none.tmpl (hunk 2 above)
Comment (rav-pradhan) [reply assets/templates/homepage/main-figures-none.tmpl]: image

Should be fixed in latest commit, with a fix for the link too

File: assets/templates/homepage/main-figures.tmpl (hunk 1 above)
Comment (rav-pradhan) [reply assets/templates/homepage/main-figures.tmpl:17]: In my most recent commit ("set fixed height...") I've set a fixed height on medium viewport only - `height-md--42` for both the employment and unemployment content containers seems to help with ensuring the layout is more aligned in that viewport. No impact to any other viewport size 

image

Comment (rav-pradhan) [reply assets/templates/homepage/main-figures.tmpl:17]: (where only one emp/unemp figure loads)
image

**Review by dandwal — APPROVED** [review]

**Pull request merged**